os/arch/arm/src/amebasmart/amebasmart_mipi.c

```diff
@@ -0,0 +1,339 @@
+	u32 vtotal, htotal_bits, bit_per_pixel, overhead_cycles, overhead_bits, total_bits;
+	priv->MIPI_InitStruct->MIPI_VideoDataFormat = MIPI_VIDEO_DATA_FORMAT_RGB565;
+	switch (priv->MIPI_InitStruct->MIPI_VideoDataFormat) {
+	case MIPI_VIDEO_DATA_FORMAT_RGB565:
+		bit_per_pixel = 16;
+		break;
+	case MIPI_VIDEO_DATA_FORMAT_RGB666_PACKED:
+		bit_per_pixel = 18;
+		break;
+	case MIPI_VIDEO_DATA_FORMAT_RGB666_LOOSELY:
+	case MIPI_VIDEO_DATA_FORMAT_RGB888:
+	default:
+		bit_per_pixel = 24;
+		break;
+	}
+	priv->MIPI_InitStruct->MIPI_LaneNum = dsi_host->config.lcd_lane_num;
+	priv->MIPI_InitStruct->MIPI_FrameRate = dsi_host->config.mipi_frame_rate;
+	priv->MIPI_InitStruct->MIPI_HSA = dsi_host->config.mipi_dsi_HSA * bit_per_pixel / 8;	//- 10; /* here the unit is pixel but not us */
+	if (priv->MIPI_InitStruct->MIPI_VideoModeInterface == MIPI_VIDEO_NON_BURST_MODE_WITH_SYNC_PULSES) {
+		priv->MIPI_InitStruct->MIPI_HBP = dsi_host->config.mipi_dsi_HBP * bit_per_pixel / 8;	//- 10;
+	} else {
+		priv->MIPI_InitStruct->MIPI_HBP = (dsi_host->config.mipi_dsi_HSA + dsi_host->config.mipi_dsi_HBP) * bit_per_pixel / 8;	//-10 ;
+	}
+	priv->MIPI_InitStruct->MIPI_HACT = dsi_host->config.XPixels;
+	priv->MIPI_InitStruct->MIPI_HFP = dsi_host->config.mipi_dsi_HFP * bit_per_pixel / 8;	//-12;
+	priv->MIPI_InitStruct->MIPI_VSA = dsi_host->config.mipi_dsi_VSA;
+	priv->MIPI_InitStruct->MIPI_VBP = dsi_host->config.mipi_dsi_VBP;
+	priv->MIPI_InitStruct->MIPI_VACT = dsi_host->config.YPixels;
+	priv->MIPI_InitStruct->MIPI_VFP = dsi_host->config.mipi_dsi_VFP;
+	vtotal = priv->MIPI_InitStruct->MIPI_VSA + priv->MIPI_InitStruct->MIPI_VBP + priv->MIPI_InitStruct->MIPI_VACT + priv->MIPI_InitStruct->MIPI_VFP;
+	htotal_bits = (dsi_host->config.mipi_dsi_HSA + dsi_host->config.mipi_dsi_HBP + priv->MIPI_InitStruct->MIPI_HACT + dsi_host->config.mipi_dsi_HFP) * bit_per_pixel;
+	overhead_cycles = T_LPX + T_HS_PREP + T_HS_ZERO + T_HS_TRAIL + T_HS_EXIT;
+	overhead_bits = overhead_cycles * priv->MIPI_InitStruct->MIPI_LaneNum * 8;
+	total_bits = htotal_bits + overhead_bits;
+
+	priv->MIPI_InitStruct->MIPI_VideDataLaneFreq = priv->MIPI_InitStruct->MIPI_FrameRate * total_bits * vtotal / priv->MIPI_InitStruct->MIPI_LaneNum / Mhz + 20;
+	priv->MIPI_InitStruct->MIPI_LineTime = (priv->MIPI_InitStruct->MIPI_VideDataLaneFreq * Mhz) / 8 / priv->MIPI_InitStruct->MIPI_FrameRate / vtotal;
+	priv->MIPI_InitStruct->MIPI_BllpLen = priv->MIPI_InitStruct->MIPI_LineTime / 2;
+
+#ifdef CONFIG_LCD_ST7785
+	if (dsi_host->config.mipi_dsi_HSA + dsi_host->config.mipi_dsi_HBP + dsi_host->config.XPixels + dsi_host->config.mipi_dsi_HFP < 250 + dsi_host->config.mipi_dsi_RTNI * 16) {
+		mipidbg("!!ERROR!!, LCM NOT SUPPORT\n");
+	}
+#elif CONFIG_LCD_ST7701
+	if (dsi_host->config.mipi_dsi_HSA + dsi_host->config.mipi_dsi_HBP + dsi_host->config.XPixels + dsi_host->config.mipi_dsi_HFP < 512 + dsi_host->config.mipi_dsi_RTNI * 16) {
+		mipidbg("!!ERROR!!, LCM NOT SUPPORT\n");
+        }
+#endif
```

Comment:
1. `#elif` should be `#elif defined( )`. Current one is wrong usage in Tizen Lite config.
2. I don't agree adding lcd specific code in a common. Is there no way to get this through a function arg? or it should be a mipi common config like CONFIG_MIPI_LCM_512 or CONFIG_MIPI_LCD_LIMIT 512 or something like that.